UnityProject/Assets/Scripts/HealthV2/Living/BodyPartMutations.cs

```diff
@@ -284,10 +387,142 @@ public void OnDestroy()
 	}
 
 
-	public void ChangeToSpecies(PlayerHealthData PlayerHealthData, GameObject BodyPart)
+	public void ChangeToSpecies(PlayerHealthData PlayerHealthData, GameObject BodyPart, CharacterSheet characterSheet = null)
 	{
 		if (this.TryGetComponent<Brain>(out var brain)) return; //Make it a little bit harder to remove from a round
-		StartCoroutine(ProcessChangeToSpecies(PlayerHealthData, BodyPart));
+		if (characterSheet != null)
+		{
+			var SpawnedBodypart = Spawn.ServerPrefab(BodyPart).GameObject.GetComponent<BodyPart>();
+
+			ColorUtility.TryParseHtmlString(characterSheet.SkinTone, out var bodyColor);
+
+			bool HasOpenProcedure = Enumerable.OfType<OpenProcedure>(SpawnedBodypart.SurgeryProcedureBase).Any();
+			var bodyParts = characterSheet.GetRaceSoNoValidation();
+
+			var bodyPartExample = bodyParts.Base.Head.Elements[0];
+
+			switch (RelatedPart.BodyPartType)
+			{
+				case BodyPartType.Head:
+					bodyPartExample = bodyParts.Base.Head.Elements[0];
+					break;
+				case BodyPartType.Chest:
+					bodyPartExample = bodyParts.Base.Torso.Elements[0];
+					break;
+				case BodyPartType.LeftArm:
+					bodyPartExample = bodyParts.Base.ArmLeft.Elements[0];
+					break;
+				case BodyPartType.RightArm:
+					bodyPartExample = bodyParts.Base.ArmRight.Elements[0];
+					break;
+				case BodyPartType.LeftLeg:
+					bodyPartExample = bodyParts.Base.LegLeft.Elements[0];
+					break;
+				case BodyPartType.RightLeg:
+					bodyPartExample = bodyParts.Base.LegRight.Elements[0];
+					break;
+			}
+
+			var bodyPartExampleStorage = bodyPartExample.GetComponent<ItemStorage>();
+			var usedOrgansInSpawnedPart = new List<GameObject>();
+
+			if (bodyPartExampleStorage.Populater.SlotContents.Count() != 0)
+			{
+				foreach (var x in bodyPartExampleStorage.Populater.SlotContents)
+				{
+					if (x.Prefab != null)
+						usedOrgansInSpawnedPart.Add(x.Prefab);
+				}
+			}
+			else
+			{
+				usedOrgansInSpawnedPart.AddRange(bodyPartExampleStorage.Populater.DeprecatedContents);
+			}
+
+			foreach (var itemSlot in RelatedPart.OrganStorage.GetItemSlots())
+			{
+				if (itemSlot.Item != null)
```

Comment:
same here, too much nesting.

Reply:
Gonna return that method back to what was before because implemented all changeling logic through other method